atox/src/main/kotlin/ui/call/CallViewModel.kt

```diff
@@ -46,6 +49,17 @@ class CallViewModel @Inject constructor(
     fun startSendingAudio() = callManager.startSendingAudio()
     fun stopSendingAudio() = callManager.stopSendingAudio()
 
+    fun toggleSpeakerphone() {
+        speakerphoneOn = !speakerphoneOn
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            if (speakerphoneOn) {
+                proximityScreenOff.release()
```

Comment:
It looks like wakelocks keep a reference counter internally (unless disabled with setReferenceCounted), so this can break, leaking a wakelock.

If a call is started with speakerphone on, toggling this twice will first increment the reference counter (so it's 2 now), and then decrease it and null out the wakelock reference in ProximityScreenOff, leading to it leaking with the reference counter being stuck at 1. (Something similar will happen if toggled once and then ending the call.)

Reply:
Does it really able to acquire it more than once after this condition?
https://github.com/evilcorpltd/aTox/blob/030195b519de03a3ded356fe516b148bcf5d06a2/atox/src/main/kotlin/ProximityScreenOff.kt#L21

Reply:
No, I must've misread or something. :P Who knows, this was months ago...